tests/test_util.py

```diff
@@ -456,3 +483,57 @@ def test_df_parser(tmpdir):
     assert len(df.labels) == 1
     assert df.labels.get('label') == 'foobar barfoo'
 
+def test_df_parser_parent_env_arg(tmpdir):
+    tmpdir_path = str(tmpdir.realpath())
+    p_env = {
+        "test_env": "first"
+    }
+    df_content = dedent("""\
+        FROM fedora
+        ENV foo=bar
+        LABEL label="foobar $test_env"
+        """)
+
+    with open(tmpdir_path + '/Dockerfile', 'w') as dockerfile:
+        dockerfile.write(df_content)
+        dockerfile.flush()
+
+    df = df_parser(tmpdir_path, parent_env=p_env)
+    assert df.labels.get('label') == 'foobar first'
+
+@pytest.mark.parametrize('env_arg', [
+    {"test_env": "first"},
+    ['test_env=first'],
+    ['test_env_first'],
+])
+def test_df_parser_parent_env_wf(tmpdir, request, env_arg):
+    tmpdir_path = str(tmpdir.realpath())
+    df_content = dedent("""\
+        FROM fedora
+        ENV foo=bar
+        LABEL label="foobar $test_env"
+        """)
+    env_conf = {INSPECT_CONFIG: {"Env": env_arg}}
+
+    with open(tmpdir_path + '/Dockerfile', 'w') as dockerfile:
+        dockerfile.write(df_content)
+        dockerfile.flush()
+
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image')
+    flexmock(workflow, base_image_inspect=env_conf)
+    fake_logger = FakeLogger()
+    existing_logger = util.logger
+    util.logger = fake_logger
+
+    def restore_logger():
+        util.logger = existing_logger
+    
+    request.addfinalizer(restore_logger)
+    df = df_parser(tmpdir_path, workflow=workflow)
+
+    if isinstance(env_arg, list) and ('=' not in env_arg[0]):
+        expected_log_message = ("Unable to parse all of Parent Config ENV")
+        assert (expected_log_message, ) in fake_logger.debugs
```

Comment:
You can use `caplog` to do this sort of thing and it's generally easier. Just add 'caplog' to the parameter list for that method and the pytest caplog fixture will be passed in -- there are other examples around in this project.

You probably want something like:
```
with caplog.atLevel(logging.DEBUG):
   ...  # use df_parser()

assert msg in [x.message for x in caplog.records()]
```